code/player/TTTPlayer.Nameplate.cs

```diff
@@ -54,28 +54,32 @@ public TTTPanel DisplayHint(TTTPlayer client)
             return new EntityHintPanel("Name", "Health")
                 .WithBackground()
                 .WithTick((display) =>
-               {
-                   if (Health == 0 && LifeState == LifeState.Alive)
-                   {
-                       display.BottomLabel.SetText(""); // network-sync workaround
-                   }
-                   else
-                   {
-                       //Set health group related data on bottom label.
-                       float health = Health / MaxHealth * 100;
-                       HealthGroup healthGroup = GetHealthGroup(health);
+                {
+                    if (this == null || !IsValid)
+                    {
+                        return;
+                    }
+
+                    if (Health == 0 && LifeState == LifeState.Alive)
```

Comment:
Do we just want to make a call to `IsSpectator` here?

Reply:
dk, just fixed a bug that's bugged the game while testing... Would it keep as it is until a rework is planned (or in Alex's branch)